os/se/security_level/security_level.c

```diff
@@ -85,17 +85,21 @@ int set_security_level(void)
 	struct sec_lowerhalf_s *se = se_get_device();
 	hal_data output;
 	hal_result_e hal_ret;
-	security_level_result_t ret;
 
 	output.priv_len = 0;
 	output.priv = NULL;
 	output.data_len = 2;
 	output.data = (char *)kmm_malloc(output.data_len);
 
+	if (output.data == NULL) {
+		kmm_free(output.data);
```

Comment:
`if (output.data == NULL)` means fail to allocate. Then, there is no address to free.